app/controllers/with_user_controller.rb

```diff
@@ -4,16 +4,29 @@ class WithUserController < ApplicationController
   before_action :authorize!
 
   def authorize!
+    # make sure we have a valid auth to begin with
+    render_unauthorized and return unless current_user_id && current_user
+
+    # all with user controllers also operate on a subject, sometimes "me".
+    # we want this here, after the token is checked, but before the subject authorization is checked
+    @user = params[user_id_attribute].blank? || params[user_id_attribute] == "me" ? current_user : User.find_by(id: params[user_id_attribute])
+
     # requested is authorized if using okta to provide a valid user id, and no specific user is set
-    return if authorization && current_user && !@user
+    return if authorization && current_user && params[user_id_attribute].blank?
 
-    # currently, if trying to operate on a specific @user, you can only operate on your own user data, but this may change later
+    # can always operate on your own user
     return if @user && @user == current_user
 
+    # currently, if trying to operate on a specific @user, you can only operate on your own user data, but this may change later
+    render_forbidden and return if @user.nil? || @user != current_user
+
     render_unauthorized
   end
```

Comment:
I believe `authorize` can be simplified to this:
```ruby
  def authorize!
    # make sure we have a valid auth to begin with
    render_unauthorized and return unless current_user_id && current_user && authorization

    # all with user controllers also operate on a subject, sometimes "me".
    # we want this here, after the token is checked, but before the subject authorization is checked
    @user = params[user_id_attribute].blank? || params[user_id_attribute] == "me" ? current_user : User.find_by(id: params[user_id_attribute])

    # currently, if trying to operate on a specific @user, you can only operate on your own user data, but this may change later
    render_forbidden and return if @user != current_user
  end
```

Reply:
When I change it out locally all the current tests pass and it looks logically correct